### PR TITLE
fix: collection picker version list NOT to mix sources

### PR DESCRIPTION
**Pull request by abhikdps**
## Description

Fix collection picker version list NOT to mix sources

## Related Issues

NA

## Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update

## Screenshots (if applicable)

Screenshot 2026-04-09 at 22 09 40
Screenshot 2026-04-09 at 22 09 55

## Checklist

- [x] Code follows project style
- [x] Tests pass locally
- [ ] Documentation updated

## Summary by CodeRabbit

* **Bug Fixes**
  * Collections from multiple sources are tracked separately with source identifiers in version entries.
  * Version deduplication now includes source, preventing cross-source collisions.
  * Version picker filters options by selected source and preserves null/placeholder version entries.

* **Tests**
  * Added coverage for multi-source collections, source-aware deduplication, and semver-with-null version scenarios.

### CODE REVIEW

**Review by NilashishC — COMMENTED** [review]
Thanks for the fix—**Comment** review (non-blocking).

The multi-source dedupe plus per-source filtering matches the bug described in the PR. Backend tests cover the catalog merge well. A few optional follow-ups in inline comments (edge case on ref-only rows, label when `version` is null, and mixed `source` metadata).

File: plugins/self-service/src/components/Scaffolder/CollectionsPicker/CollectionsPickerExtension.tsx [hunk 1]
```diff
@@ -210,15 +210,22 @@ export const CollectionsPickerExtension = ({
...
+        ? details.filter((d: SourceVersionDetail) => d.source === sourceId)
+        : details;
       const stringsOnly = rawVersions.filter(
         (v: unknown): v is string => typeof v === 'string',
       );
       const stringsForSource: string[] | undefined =
         foundCollection?.sourceVersions?.[sourceId];
 
-      if (stringsForSource?.length && details.length) {
+      if (stringsForSource?.length && detailsForSource.length) {
```
[the remark this answers is not in the document]
Comment (abhikdps) [reply plugins/self-service/src/components/Scaffolder/CollectionsPicker/CollectionsPickerExtension.tsx]: fixed in [9cb4d06](https://github.com/ansible/ansible-backstage-plugins/pull/241/commits/9cb4d06f0caf478fbca82d5f9c9b1867981147cb)

**Review by NilashishC — APPROVED** [review]
